### ISSUE TITLE
VM templates created by New-avdVmTemplate don't show up via Get-avdHostPool | select vmtemplate

### ISSUE THREAD

**Issue by MohnJadden** [srozemuller/AzAvd#18]
I'm following through the steps to add a [VM template](https://rozemuller.com/using-vm-templates-for-avd-sessionhost-automation/#add-vm-template).  This is for an existing host pool with hosts already in it.  I am running the following cmdlet as a user with the Contributor role: 

`New-avdvmtemplate -hostpoolname Company-vdi-Infrastructure-hostpool-eastus -ResourceGroupName Company-vdi-Infrastructure-rg-eastus -namePrefix infra-v -vmSku 'Standard_B2ms' -vmCores 2 -vmRam 8 -osdisktype standardssd_lrs`

I get the following output:

```
name       : Company-vdi-Infrastructure-hostpool-eastus
id         : /subscriptions/guid/resourcegroups/Company-vdi-Infrastructure-rg-eastus/providers/Microsoft.DesktopVirtualization/hostpools/Company-vdi-Infrastructure-hostpool-eastus
type       : Microsoft.DesktopVirtualization/hostpools
location   : eastus
tags       : @{DepartmentName=IT - Support; Location=NY; Owner=boss,; TechnicalContact=me,}
kind       :
systemData : @{createdBy=me@company.com; createdByType=User; createdAt=2022-02-22T14:41:20.13Z; lastModifiedBy=me@company.com; lastModifiedByType=User;
             lastModifiedAt=2022-03-10T19:49:59.23Z}
properties : @{friendlyName=Infrastructure VDI; description=Company Infrastructure team VDI; hostPoolType=Personal; personalDesktopAssignmentType=Automatic; applicationGroupReferences=System.Object[]; customRdpProperty=drivestoredirect:s:;audiomode:i:0;videoplaybackmode:i:1;redirectclipboard:i:1;redirectprinters:i:0;devicestoredirect:s:;redirectcomports:i:0;redirectsmartcards:i:0;usbdevicestoredirect:s:;enablecredsspsupport:i:1;use multimon:i:1;maximizetocurrentdisplays:i:1;singlemoninwindowedmode:i:1;screen mode id:i:1;smart sizing:i:1;desktop sizeid:i:3;desktopheight:i:1024;desktopwidth:i:1280;targetisaadjoined:i:1;audiocapturemode:i:1;camerastoredirect:s:*;bandwidthautodetect:i:1;networkautodetect:i:1;compression:i:1;;
             maxSessionLimit=999999; loadBalancerType=Persistent; validationEnvironment=False; ring=; registrationInfo=; vmTemplate={"domain":"","osDiskType":"standardssd_lrs","namePrefix":"infra-v"," 
             vmSize":{"cores":"2","ram":"8","id":"Standard_B2ms"},"galleryImageOffer":"","galleryImagePublisher":"","galleryImageSKU":"","imageType":"","imageUri":"","customImageId":"","useManagedDisk 
             s":"True","galleryItemId":null}; preferredAppGroupType=Desktop; migrationRequest=; cloudPcResource=False; startVMOnConnect=True; ssoadfsAuthority=; ssoClientId=;
             ssoClientSecretKeyVaultPath=; ssoSecretType=; objectId=anotherguid}
```
Afterwards, when I run `Get-avdHostPool -hostpoolname Company-vdi-Infrastructure-hostpool-eastus -ResourceGroupName Company-vdi-Infrastructure-rg-eastus` , I only get the following output:

```
vmtemplate
----------
```
There's nothing to confirm if the template actually was changed as intended, nor if it's usable for future Az.Avd automation.  It would be useful to have some kind of output that indicates what actually occurred when New-avdVmTemplate is run, or at least its success/failure.

**Comment by srozemuller** [comment]
@MohnJadden I have tested the VM template command, but got no issues.
When looking in your properties output, I see the vmtemplate is filled.

Good to know is that the vmTemplate information is stored under properties. So if you want to get the template, store the output into an variable and browse $var.properties.vmtemplate.

good point about the successful return. I will update the code for that.

**Comment by srozemuller** [comment]
The code is updated and available in the v2.1.4-beta.1 release
